app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+
+  def create
+    @user = User.find_by user_name: params[:user][:user_name]
+
+
+    if @user
+      session[:user_id] = @user.id
+    else
+      @user = User.create user_name: params[:user][:user_name]
```

Comment:
You should check to verify that `create` worked!  I tried to log in with a blank username and got no error message.